vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SearchManager.java

```diff
@@ -0,0 +1,364 @@
+
+                if (slide != null && slide.getFirstImageBlock() != null) {
+                    firstSlideImageId = slide.getFirstImageBlock().getImage().getId();
+                }
+                moduleFirstSlideImage.put(module.getId(), firstSlideImageId);
+            }
+
+        }
+        searchModuleResults.setModuleImageIdMap(moduleFirstSlideImage);
+        searchModuleResults.setModuleAlertMessages(isModuleConfiguredMap);
+
+        return searchModuleResults;
+    }
+
+    protected SearchSlideResults convertToSearchSlideResults(List<ISlide> slideList) {
+        SearchSlideResults searchSlideResults = new SearchSlideResults();
+        searchSlideResults.setSlides(slideList);
+
+        Map<String, String> slideFirstImage = new HashMap<>();
+
+        for (ISlide slide : slideList) {
+            if (slide != null && slide.getFirstImageBlock() != null) {
+                slideFirstImage.put(slide.getId(), slide.getFirstImageBlock().getImage().getId());
+            }
+        }
+        searchSlideResults.setFirstImageOfSlide(slideFirstImage);
+        return searchSlideResults;
+    }
+
+    protected SearchSlideTextBlockResults convertToSearchSlideTextBlockResults(List<ISlide> slideTextList, String searchTerm) {
+        SearchSlideTextBlockResults staffSearchSlideTextBlockResults = new SearchSlideTextBlockResults();
+        staffSearchSlideTextBlockResults.setSlidesWithMatchedTextBlock(slideTextList);
+
+        Map<String, String> slideTextFirstImageMap = new HashMap<>();
+        Map<String, String> slideTextFirstTextBlockMap = new HashMap<>();
+
+        for (ISlide slide : slideTextList) {
+            if (slide != null) {
+                if (slide.getFirstImageBlock() != null) {
+                    slideTextFirstImageMap.put(slide.getId(), slide.getFirstImageBlock().getImage().getId());
+                }
+                if (slide.getFirstMatchedTextBlock(searchTerm) != null) {
+                    slideTextFirstTextBlockMap.put(slide.getId(), slide.getFirstMatchedTextBlock(searchTerm).htmlRenderedText());
+                }
+            }
+        }
+        staffSearchSlideTextBlockResults.setSlideToFirstImageMap(slideTextFirstImageMap);
+        staffSearchSlideTextBlockResults.setSlideToFirstTextBlockMap(slideTextFirstTextBlockMap);
+
+        return staffSearchSlideTextBlockResults;
+    }
+
+    /**
+     * This method is used to search the searched string specified in the input
+     * parameter(searchTerm) and return the published spaces corresponding to
+     * the page number specified in the input parameter(spacePagenum) whose name or
+     * description contains the search string.
+     * 
+     * @param spacePagenum current page number sent as request parameter in the URL.
+     * @param searchTerm   This is the search string which is being searched.
+     */
+    @Override
+    public SearchSpaceResults searchForSpace(String spacePagenum, String searchTerm) {
```

Comment:
why does this and the following methods not do anything (except calling another method)? That seems unnecessary.